network/common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -121,4 +123,69 @@ private static boolean isSymlink(File file) throws IOException {
     }
     return !fileInCanonicalDir.getCanonicalFile().equals(fileInCanonicalDir.getAbsoluteFile());
   }
+
+  private static ImmutableMap<String, TimeUnit> timeSuffixes = 
+    ImmutableMap.<String, TimeUnit>builder()
+      .put("us", TimeUnit.MICROSECONDS)
+      .put("ms", TimeUnit.MILLISECONDS)
+      .put("s", TimeUnit.SECONDS)
+      .put("min", TimeUnit.MINUTES)
+      .put("h", TimeUnit.HOURS)
+      .put("d", TimeUnit.DAYS)
+      .build();
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a time count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+  private static long parseTimeString(String str, TimeUnit unit) throws NumberFormatException {
+    String timeError = "Time must be specified as seconds (s), " +
+      "milliseconds (ms), microseconds (us), minutes (min) hour (h), or day(d). " +
+      "E.g. 50s, 100ms, or 250us.";
+    String lower = str.toLowerCase().trim();
+    
+    try {
+      String suffix = "";
+      for (String tail: timeSuffixes.keySet()) {
+        if (lower.endsWith(tail)) {
+          // Handle special case for overlapping suffixes
+          if (!tail.equals("s") || 
+                  (tail.equals("s") && !lower.endsWith("us") && !lower.endsWith("ms")))
+          {
+            suffix = tail;
+          }
+        }
+      }
+      
+      return unit.convert(Long.parseLong(str.substring(0, str.length() - suffix.length())),
+        timeSuffixes.containsKey(suffix) ? timeSuffixes.get(suffix) : unit);
+    } catch (NumberFormatException e) {
+      throw new NumberFormatException(timeError + "\n" + e.getMessage());
+    }
+  }
+
+  /**
+   * Convert a time parameter such as (50s, 100ms, or 250us) to microseconds for internal use. If
+   * no suffix is provided, the passed number is assumed to be in us.
+   */
+  public static long timeStringAsUs(String str) throws NumberFormatException {
```

Comment:
I just wanted to add the additional error message clarifying appropriate suffixes which is why I handle the error.